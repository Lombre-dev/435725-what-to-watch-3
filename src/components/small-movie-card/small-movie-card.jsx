import PropTypes from 'prop-types';
import React from 'react';
import {Movie} from '../types';

export default class SmallMovieCard extends React.PureComponent {

  constructor(props) {
    super(props);

    this._handleComponentHover = this._handleComponentHover.bind(this);
    this._handleTitleClick = this._handleTitleClick.bind(this);
  }

  _handleComponentHover() {

    const {movie, onComponentHoverCallback} = this.props;

    onComponentHoverCallback({movie});
  }

  _handleTitleClick(e) {

    const {movie, onTitleClickCallback} = this.props;

    e.preventDefault();
    onTitleClickCallback({movie});
  }

  render() {

    const {movie} = this.props;

    return (
      <article className="small-movie-card catalog__movies-card" onMouseEnter={this._handleComponentHover}>
        <div className="small-movie-card__image">
          <img src={movie.frames[0]} alt={movie.title} width="280" height="175" />
        </div>
        <h3 className="small-movie-card__title">
          <a className="small-movie-card__link" href="movie-page.html" onClick={this._handleTitleClick}>{movie.title}</a>
        </h3>
      </article>
    );
  }
}

SmallMovieCard.propTypes = {
  movie: Movie.isRequired,
  onTitleClickCallback: PropTypes.func.isRequired,
  onComponentHoverCallback: PropTypes.func.isRequired,
};
