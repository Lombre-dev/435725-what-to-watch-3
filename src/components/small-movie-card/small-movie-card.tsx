import * as React from 'react';
import {Link} from 'react-router-dom';
import {AppPages, PlayerState} from '../../consts';
import {TMovie} from '../types';
import VideoPlayer from '../video-player/video-player';

type TSmallMovieCardProps = {
  movie: TMovie;
  isPreviewActive: boolean;
  onHover: Function;
  onLeave: Function;
};

class SmallMovieCard extends React.PureComponent<TSmallMovieCardProps> {
  public constructor(props: TSmallMovieCardProps) {
    super(props);

    this._handleHover = this._handleHover.bind(this);
    this._handleLeave = this._handleLeave.bind(this);
  }

  private _handleHover() {

    const {movie, onHover} = this.props;

    onHover({id: movie.id});
  }

  private _handleLeave() {

    const {movie, onLeave} = this.props;

    onLeave({id: movie.id});
  }

  public render() {

    const {movie, isPreviewActive} = this.props;

    return (
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={this._handleHover}
        onMouseLeave={this._handleLeave}
      >
        <Link className="small-movie-card__link" to={`${AppPages.MOVIES}/${movie.id}`}>
          <div className="small-movie-card__image" >
            <VideoPlayer
              id={movie.id}
              state={isPreviewActive ? PlayerState.PLAYING : PlayerState.LOADING}
              poster={movie.poster}
              src={movie.preview}
            />
          </div>
          <h3 className="small-movie-card__title">{movie.title}</h3>
        </Link>
      </article>
    );
  }
}

export default SmallMovieCard;
