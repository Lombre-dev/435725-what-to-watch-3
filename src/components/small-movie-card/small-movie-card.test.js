import React from 'react';
import renderer from 'react-test-renderer';
import {SmallMovieCard} from './small-movie-card';

const ID = 0;
const MOVIE = {
  title: `The Grand Budapest Hotel`,
  genres: [`Drama`],
  year: 2014,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  frames: [`img/the-grand-budapest-hotel-poster.jpg`],
  ratingScore: 8.9,
  ratingReviewsCount: 240,
  preview: `./samples/sintel_trailer-480p.mp4`,
  src: `./samples/sintel_trailer-480p.mp4`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  story: `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  duration: 99,
  actors: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Some Actor 1`,
    `Some Actor 2`,
  ],
  reviews: [
    {
      author: `Some Reviewer`,
      score: 8.2,
      text: `Awesome text ...`,
      date: 1582590140667,
    }
  ]
};
const IS_PREVIEW_ACTIVE = false;
const HANDLE_EVENT = () => {};

describe(`<SmallMovieCard />`, () => {

  it(`render should be match markup`, () => {

    const result = renderer
      .create(<SmallMovieCard
        id={ID}
        movie={MOVIE}
        isPreviewActive={IS_PREVIEW_ACTIVE}
        onHover={HANDLE_EVENT}
        onLeave={HANDLE_EVENT}
        onClick={HANDLE_EVENT}
      />, {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
