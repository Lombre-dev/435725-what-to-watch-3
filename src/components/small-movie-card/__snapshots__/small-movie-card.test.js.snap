// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallMovieCard /> render should be match markup 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwal"
      height="175"
      src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Fantastic Beasts: The Crimes of Grindelwal
    </a>
  </h3>
</article>
`;
